pkg/reconciler/pipeline/dag/dag_test.go

```diff
@@ -456,24 +423,15 @@ func TestBuild_InvalidDAG(t *testing.T) {
 		RunAfter: []string{"d"},
 	}
 	gDependsOnF := v1beta1.PipelineTask{
-		Name: "g",
-		Resources: &v1beta1.PipelineTaskResources{
-			Inputs: []v1beta1.PipelineTaskInputResource{{From: []string{"f"}}},
-		},
+		Name:     "g",
+		RunAfter: []string{"f"},
 	}
 
 	tcs := []struct {
 		name string
 		spec v1beta1.PipelineSpec
 		err  string
 	}{{
-		// a
-		// |
-		// a ("a" depends on resource from "a")
-		name: "self-link-from",
```

Comment:
is there another test that detects cycles using runAfter?

Reply:
Yes there is and i replaced this one with results and params as @afrittoli suggested.